zerver/lib/send_email.py

```diff
@@ -532,12 +532,13 @@ def send_custom_email(
     rendered_input = render_markdown_path(plain_text_template_path.replace("templates/", ""))
 
     # And then extend it with our standard email headers.
-    with open(html_source_template_path, "w") as f:
-        with open(markdown_email_base_template_path) as base_template:
-            # Note that we're doing a hacky non-Jinja2 substitution here;
-            # we do this because the normal render_markdown_path ordering
-            # doesn't commute properly with inline_email_css.
-            f.write(base_template.read().replace("{{ rendered_input }}", rendered_input))
+    with open(html_source_template_path, "w") as f, open(
+        markdown_email_base_template_path
+    ) as base_template:
```

Comment:
This simplification rule is nice for the cases where it doesn't adjust line-wrap, but in this class of case, where previously we had a `with` statement on a single line, and the result of this rule is that it's split across 3, I feel like it's a regression. 

I'm not sure if there's a way to make the rule depend on it, but I suppose we could do it manually and not turn on the rule.